test/gpu/octree/test_radius_search.cpp

```diff
@@ -125,142 +183,17 @@ TEST(PCL_OctreeGPU, batchRadiusSearch)
-            ASSERT_NEAR (sqr_distances_batch[j], sqr_dist_host[j], 0.001);
-        }
-        ASSERT_EQ ( ( results_batch == results_host ), true );       
-    }    
-
-    const float avg_size_individual = std::accumulate(sizes_individual.begin(), sizes_individual.end(), 0) * (1.f/sizes_individual.size());
-
-    std::cout << "avg_result_size_individual = " << avg_size_individual << "\n";
-    ASSERT_GT(avg_size_individual, 5);
-
-
-    //verify results    
-    for(std::size_t i = 0; i < data.tests_num; i+=2)
-    {                
-        std::vector<int>& results_host = host_search_shared[i];
-        std::vector<float>& sqr_dist_host = host_sqr_distances_shared[i];
-
-        const int beg = i/2 * max_answers;
-        const int end = beg + sizes_shared_indices[i/2];
-
-        const auto beg_dist2 = sqr_distances_shared_indices.cbegin() + beg;
-        const auto end_dist2 = sqr_distances_shared_indices.cbegin() + end;
-
-        std::vector<int> results_batch (downloaded_buffer_shared_indices.begin() + beg, downloaded_buffer_shared_indices.begin() + end);
-        const std::vector<float> sqr_distances_batch (beg_dist2, end_dist2);
-
-        std::sort(results_batch.begin(), results_batch.end());
-        std::sort(results_host.begin(), results_host.end());
-
-        if ((int)results_batch.size() == max_answers && results_batch.size() < results_host.size() && max_answers)
-        {
-            results_host.resize(max_answers);
-            sqr_dist_host.resize(max_answers);
-        }
-
-        ASSERT_EQ ( ( results_batch == results_host ), true );       
-
-        const float sqr_radius = data.shared_radius * data.shared_radius;
-        for (std::size_t j = 0; j < sqr_dist_host.size(); j++)
-        {
-            ASSERT_LT (sqr_distances_batch[j], sqr_radius);
-            ASSERT_NEAR (sqr_distances_batch[j], sqr_dist_host[j], 0.001);
-        }
-    }
+    //verify results
+    const std::vector<float> radiuses (data.tests_num, data.shared_radius);
+    std::cout << "shared radius test : ";
+    verifyResults(host_search_shared, host_sqr_distances_shared, result_device_shared, result_sqr_distances_shared, radiuses, max_answers, data.tests_num, 1);
 
-    const float avg_size_shared_indices = std::accumulate(sizes_shared_indices.begin(), sizes_shared_indices.end(), 0) * (1.f/sizes_shared_indices.size());
+    std::cout << "individual radius test : ";
+    verifyResults(host_search_individual, host_sqr_distances_individual, result_device_individual, result_sqr_distances_individual, data.radiuses, max_answers, data.tests_num, 1);
 
-    std::cout << "avg_result_size_shared_indices = " << avg_size_shared_indices << "\n";
-    ASSERT_GT(avg_size_shared_indices, 5);
+    std::cout << "shared radius with indices test : ";
+    verifyResults(host_search_shared, host_sqr_distances_shared, result_device_shared_indices, result_sqr_distances_shared_indices, radiuses, max_answers, data.tests_num, 2);
```

Comment:
You should convert this test to a fixture, with the three test cases. that wy google-test does this status printing for you, instead of you being the one printing which check you're running.